From: Larry Finger <Larry.Finger@lwfinger.net>
Subject: Possible fix for firmware problem with RTL8192DU

This device comes in 3 varieties: (1) Single MAC - Single PHY, (2) Double MAC - 
Single PHY, and (3) Double MAC - Double PHY.

The Single MAC version never loads the firmware from that data array.
It is never explicitly stated in a comment, but apparently the fw comes
from some kind of EEPROM on the device.

From: Larry Finger <Larry.Finger@lwfinger.net>
Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>

--- a/drivers/staging/rtl8192du/hal/rtl8192d_hal_init.c
+++ b/drivers/staging/rtl8192du/hal/rtl8192d_hal_init.c
@@ -331,6 +331,10 @@ int FirmwareDownload92D(struct rtw_adapt
 	if (adapter->bSurpriseRemoved)
 		return _FAIL;
 
+	/* Single MAC Single PHY units break if external firmware is loaded */
+	if (pHalData->MacPhyMode92D == SINGLEMAC_SINGLEPHY)
+		return rtStatus;
+
 	if (!adapter->firmware || !adapter->firmware->buffer) {
 		if (!get_fw_from_file(adapter)) {
 			rtStatus = _FAIL;
